Add delete resource command

16yuki0702 opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes
Add deleting the pipelineresource feature for https://github.com/tektoncd/cli/issues/193.
Make PR not too big, I started only pipelineresource.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

<!--
Does your PR contain User facing changes?

If so, briefly describe them here so we can include this description in the
release notes for the next release!
-->

```
you can delete pipelineresource from tkn.
```

hrishin commented:
/ok-to-test

16yuki0702 commented:
/retest

16yuki0702 commented:
/test pull-tekton-cli-unit-tests

sthaha commented on pkg/cmd/pipelineresource/delete.go:
```diff
@@ -0,0 +1,66 @@
+
+	c := &cobra.Command{
+		Use:          "delete",
+		Aliases:      []string{"del"},
```
how about `rm` as well to match `ls` in `list`

16yuki0702 replied:
Thank you very much!!
I didn't realize the list aliases are like that, I'll match it.

sthaha commented on pkg/cmd/pipelineresource/delete.go:
```diff
@@ -0,0 +1,66 @@
+	}
+
+	if err := cs.Tekton.TektonV1alpha1().PipelineResources(p.Namespace()).Delete(preName, &metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("Failed to delete pipelineresource %q", preName)
```
we may want to provide a bit more details as to why the deletion failed by printing the `err` as well.

16yuki0702 replied:
Totally agree, thanks!!

16yuki0702 commented:
@sthaha Thank you for your review! 
I fixed it and add annotations like this PR https://github.com/tektoncd/cli/pull/226

16yuki0702 commented:
@sthaha oh... Thank you so much! I'll add the documents.

16yuki0702 commented:
@sthaha I added the documents, thanks!!
when I run ```make docs``` and ```make man```, other documents are also modified.
for now I include these into the PR.

16yuki0702 commented:
@sthaha @vdemeester Thank for reviews!!
